### ISSUE TITLE
GitLab Direct Support

### ISSUE THREAD

**Issue by operatorequals** [operatorequals/httpimport#5]
Given that the URLs of the raw files in a GitLab repo follows the format:
```
https://gitlab.com/{user}/{repo_name}/raw/{commit_hash/branch_name}/{directory/}*{file_name}
```
a `gitlab_repo()` context can be created, similar to already implemented `github_repo()`.
Arguments can be exactly the same as in `github_repo()`:
 * username
 * repo name
 * branch/commit

The context can be tested with the below Python project:
`https://gitlab.com/BananenBernd/basictools`

**Comment by operatorequals** [comment]
Also, as *GitLab* (unlike *GitHub* and *BitBucket*), can be **self hosted**, an extra argument of `domain` may be useful, to pass the location of the *GitLab* installation (defaulting to `gitlab.com`).

Example:
```python
>>> with gitlab_repo('BananenBernd', 'basictools') :
...     import basictools
```
will import from `gitlab.com`, but
```python
>>> with gitlab_repo('myname, 'myrepo', domain = '127.0.0.1:8080') :
...     import myrepo
```
will import `myrepo` package, from the *GitLab* installation at `127.0.0.1:8080`.

**Comment by operatorequals** [comment]
The feature has been implemented in `0.5.17` which is already in `master`.
```bash
python remote_git_tests.py 
[+] Importing from GitLab
<module 'gitlab.const' (<httpimport.HttpImporter object at 0x7f4be96a1e80>)>
[+] Importing from GitHub
<module 'covertutils' (<httpimport.HttpImporter object at 0x7f4be95cc0b8>)>
[+] Importing from BitBucket
<module 'pybitbucket' (<httpimport.HttpImporter object at 0x7f4be95f22e8>)>
```
